source/include/shadow.h

```diff
@@ -217,119 +245,153 @@ typedef enum ShadowStatus
  */
 #define SHADOW_THINGNAME_LENGTH_MAX    ( 128U )
 
+/**
+ * @ingroup shadow_constants
+ * @brief The maximum length of Shadow Name.
+ */
+#define SHADOW_NAME_LENGTH_MAX         ( 64U )
+
+/**
+ * @ingroup shadow_constants
+ * @brief The name string for the un-named "Classic" shadow.
+ */
+#define SHADOW_NAME_CLASSIC            ""
```

Comment:
Classic shadow doesn't have a name. This MACRO doesn't seems to have been used in the code as well. Please remove if not required.

Reply:
This macro is provided for the user. It's used by the demo and the tests. I'm happy to delete it from here, but then the user has to define it (or pass the "" literal to the macros). I made the unnamed/Classic name be an empty string because of how the console works (and because it's nice for the macros that construct the topic strings):

![image](https://user-images.githubusercontent.com/3505720/102709648-51ab6e00-42e7-11eb-9923-853a98504c76.png)

Reply:
With the backwards-compatibility work, this macro is now used extensively within shadow.h. As well as being available for the user. The backwards-compatible macros are essentially examples of how to use it.